src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
+
+    response.status(status).json(APIResponse.create(0, status, error));
+  }
```

Comment:
PullProofAI

Consider handling potential errors more gracefully and improving error message logging.
```typescript
import {
  ArgumentsHost,
  Catch,
  ExceptionFilter,
  HttpException,
  Logger,
} from '@nestjs/common';
import { Response } from 'express';
import { APIResponse } from 'src/utils/apiResponse.utils';

@Catch(HttpException)
export class HttpExceptionFilter implements ExceptionFilter {
  private logger = new Logger();

  catch(exception: HttpException, host: ArgumentsHost) {
    const context = host.switchToHttp();
    const response = context.getResponse<Response>();
    const status = exception.getStatus();
    
    let error = exception.getResponse();
    if (typeof error === 'object' && 'message' in error) {
      error = error.message.join(', ');
    } else if (typeof error === 'string') {
      error = error;
    } else {
      error = JSON.stringify(error);
    }

    this.logger.error(
      `Message : ${error}`,
      `${context.getRequest().method} ${context.getRequest().url}`,
    );

    response.status(status).json(APIResponse.create(0, status, error));
  }
}
```
In the updated code:
- Improved error handling logic for better readability and maintainability.
- Simplified error message extraction and formatting.
- Removed unnecessary variable assignments for error handling.


<!-- This is an auto-generated comment by PullProofAI -->